### PR TITLE
Make MultiCursor  more flexible

### PR DESCRIPTION
**Pull request by weihaipy**
## PR Summary
Make MultiCursor could control draw one horizontal and/or vertical line in axes under the mouse only.
It will  implement cursor in multi plot like MetaTrader4.
## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/widgets.py [hunk 1]
```diff
@@ -1447,6 +1447,77 @@ def _update(self):
...
+    between multiple axes.
+
+    Tt's similar to MultiCursor,but only one horizontal line in figure.
+
+    For the cursor to remain responsive you must keep a reference to
+    it.
+
+    Example usage::
+
+        from matplotlib.widgets import MultiAxesCrossCursor
```
Comment (jklymak) [inline lib/matplotlib/widgets.py]: Can you make this example not depend on pylab? Otherwise this looks good to me.

Comment (xiazhujie) [reply lib/matplotlib/widgets.py]: This cursor and example is based on MultiCursor,the only difference is horizontal line.

Comment (xiazhujie) [reply lib/matplotlib/widgets.py]: I have changed example not depend on pylab yet.

File: lib/matplotlib/widgets.py [hunk 2]
```diff
@@ -1332,37 +1332,47 @@ class MultiCursor(Widget):
     Provide a vertical (default) and/or horizontal line cursor shared between
     multiple axes.
 
-    For the cursor to remain responsive you must keep a reference to
-    it.
+    You can turn on/off one horizontal or one vertical in axes under the mouse
```
Comment (jklymak) [inline lib/matplotlib/widgets.py]: Thanks for the fix on the example.  But I don't understand this new option from the docstring.   What is the use case here?   It seems that for the oneVert/Horiz you are redrawing the line each time.  That doesn't seem very efficient compared to just setting the x-data...

Comment (xiazhujie) [reply lib/matplotlib/widgets.py]: You are right,i will fix it later

Comment (xiazhujie) [reply lib/matplotlib/widgets.py]: I have fixed it, and add more usage example in docstring.

**Comment by jklymak** [comment]
Not sure if we want to go ahead w/ this one.  It seems futzy, and not super useful (does it *really* hurt to have the cursor on other axes?)  But happy to be told its really useful and that I'm wrong.

**Comment by jklymak** [comment]
Thanks a lot for the contribution.  I'm going to close this as having had no action for quite a while.  OTOH, if there is still a community that wants this, please feel free to re-open, with my apologies!

**Pull request closed without merging**